pkg/relayer/proxy/relay_verifier.go

```diff
@@ -128,3 +136,33 @@ func (rp *relayerProxy) VerifyRelayRequest(
 
 	return ErrRelayerProxyInvalidSupplier
 }
+
+// getTargetSessionBlockHeight returns the block height at which the session
+// for the given relayRequest should be processed. If the session is in the
+// grace period, the session's end block height is returned. Otherwise,
+// the current block height is returned.
+// If the session has expired, then return an error.
+func (rp *relayerProxy) getTargetSessionBlockHeight(
+	ctx context.Context,
+	relayRequest *types.RelayRequest,
+) (sessionBlockHeight int64, err error) {
+	currentBlockHeight := rp.blockClient.LastNBlocks(ctx, 1)[0].Height()
+	sessionEndblockHeight := relayRequest.Meta.SessionHeader.GetSessionEndBlockHeight()
+
+	// Check if the `RelayRequest`'s session has expired.
+	if sessionEndblockHeight < currentBlockHeight {
+		// Do not process the `RelayRequest` if the session has expired and the current
+		// block height is past the session's grace period.
+		if sessionEndblockHeight < currentBlockHeight-sessionkeeper.SessionGracePeriod {
```

Comment:
Why don't we use the `checkGracePeriodEnd` helper you introduce in the other file?